src/AzureAuth.Test/CommandMainTest.cs

```diff
@@ -384,6 +388,119 @@ public void TestPromptHintPrefixWithoutOption()
+            subject.CacheFilePath.Should().Be("Z:\\normal.cache");
+        }
+
+        /// <summary>
+        /// The test to evaluate a relative cache file name in enviroment variables.
+        /// </summary>
+        [Test]
+        [Platform("Win")] // Only valid on Windows
+        public void TestCacheFileOptionWithNormalFilePathFromEnv()
+        {
+            string filenameFromEnv = "C:\\test\\absolute_from_env.cache";
+            this.envMock.Setup(env => env.Get("AZUREAUTH_CACHE_FILE")).Returns(filenameFromEnv);
+
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+        }
+
+        /// <summary>
+        /// The test to evaluate the cache file name when both enviroment variable and option exist.
+        /// Ideally use option first.
+        /// </summary>
+        [Test]
+        [Platform("Win")] // Only valid on Windows
+        public void TestCacheFileOptionWithFilenameFromEnvAndOption()
+        {
+            string filenameFromEnv = "C:\\test\\absolute_from_env.cache";
+            this.envMock.Setup(env => env.Get("AZUREAUTH_CACHE_FILE")).Returns(filenameFromEnv);
+
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.CacheFilePath = "C:\\test\\absolute_from_option.cache";
+
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.CacheFilePath.Should().Be("C:\\test\\absolute_from_option.cache");
+        }
+
+        /// <summary>
+        /// The test to evaluate a default cache file name.
+        /// </summary>
+        [Test]
+        [Platform("Win")] // Only valid on Windows
+        public void TestCacheFileOptionWithNoParameter()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+
+            string appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            string absolutePath = Path.Combine(appData, ".IdentityService", $"msal_{subject.Tenant}.cache");
+            string expected = absolutePath;
+
+            subject.CacheFilePath.Should().Be(expected);
+        }
+
+        /// <summary>
+        /// The test to evaluate a relative cache path.
+        /// </summary>

```

Comment:
The <Summary> blocks in this test file should no be required, so I would only add them if they are adding more description than is visible from the test name itself.

Reply:
I think @kyle-rader means to remove the summary.

Reply:
The summary is required here. I added more description instead.